### ISSUE TITLE
[fix] Modal doesn't work with keyboard using JAWS

### ISSUE THREAD

**Issue by jim** [trussworks/react-uswds#1886]
**ReactUSWDS Version & USWDS Version:**

Latest version of ReactUSWDS. Does not occur with USWDS.

**Describe the bug**

When a user is using Chrome and JAWS, the content within modals (including buttons) can't be interacted with using the keyboard.

**To Reproduce**

1. Start Chrome and JAWS
2. Go to [the modal page in Storybook](https://trussworks.github.io/react-uswds/?path=/story/components-modal--default-modal)
3. Open the modal
4. Attempt to close the modal using the buttons within it

**Expected behavior**

The buttons should work just as they do when JAWS isn't being used.

**Additional context**

I think the issue is that we're adding `aria-hidden` to parent elements of the modal, and that this removes the modal's content from the accessibility tree. The original USWDS JS assumes that modals exist outside the normal flow of the page, so this technique isn't an issue there.
